src/java/org/apache/cassandra/schema/SchemaTransformation.java

```diff
@@ -37,7 +37,7 @@ public interface SchemaTransformation
     /**
      * Apply a statement transformation to a schema snapshot.
      * <p>
-     * Implementing methods should be side-effect free (outside of throwing exceptions if the transformation cannot
+     * Implementing methods should be side effect free (outside of throwing exceptions if the transformation cannot
```

Comment:
`side-effect` is also a correct spelling: https://www.collinsdictionary.com/dictionary/english/side-effect.

I would say `side-effect` but `side-effect free`.